src/plugins/controls/public/control_group/external_api/control_group_renderer.tsx

```diff
@@ -6,136 +6,186 @@
  * Side Public License, v 1.
  */
 
-import { isEqual, pick } from 'lodash';
-import React, {
-  forwardRef,
-  useEffect,
-  useImperativeHandle,
-  useMemo,
-  useRef,
-  useState,
-} from 'react';
+import { omit } from 'lodash';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { BehaviorSubject, Subject } from 'rxjs';
 import { v4 as uuidv4 } from 'uuid';
 
-import { EmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import { ReactEmbeddableRenderer, ViewMode } from '@kbn/embeddable-plugin/public';
 import type { Filter, Query, TimeRange } from '@kbn/es-query';
-import { compareFilters } from '@kbn/es-query';
+import { useSearchApi, type ViewMode as ViewModeType } from '@kbn/presentation-publishing';
 
+import { CONTROL_GROUP_TYPE } from '../../../common';
 import {
-  getDefaultControlGroupInput,
-  getDefaultControlGroupPersistableInput,
-  persistableControlGroupInputKeys,
-} from '../../../common';
-import { ControlGroupContainer } from '../embeddable/control_group_container';
-import { ControlGroupContainerFactory } from '../embeddable/control_group_container_factory';
+  ControlGroupApi,
+  ControlGroupRuntimeState,
+  ControlGroupSerializedState,
+} from '../../react_controls/control_group/types';
 import {
-  ControlGroupCreationOptions,
-  ControlGroupInput,
-  ControlGroupOutput,
-  CONTROL_GROUP_TYPE,
-} from '../types';
-import {
-  AwaitingControlGroupAPI,
-  buildApiFromControlGroupContainer,
-  ControlGroupAPI,
-} from './control_group_api';
-import { controlGroupInputBuilder, ControlGroupInputBuilder } from './control_group_input_builder';
+  controlGroupStateBuilder,
+  type ControlGroupStateBuilder,
+} from '../../react_controls/control_group/utils/control_group_state_builder';
+import { getDefaultControlGroupRuntimeState } from '../../react_controls/control_group/utils/initialization_utils';
+import { ControlGroupCreationOptions, ControlGroupRendererApi } from './types';
 
 export interface ControlGroupRendererProps {
-  filters?: Filter[];
+  onApiAvailable: (api: ControlGroupRendererApi) => void;
```

Comment:
My feeling was, no - why would you want a `ControlGroupRenderer` without the API? Because then you can't even respond to filters changing or anything.... To me, the only benefit to making it optional would be to make mocks slightly easier, but I'm not sure if that is a good enough argument? What do you think?

Reply:
Our reasoning makes sense and requiring the props is fine.